src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/Unsafe.cs

```diff
@@ -33,6 +33,7 @@ public static unsafe partial class Unsafe
         [CLSCompliant(false)]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static void* AsPointer<T>(ref T value)
```

Comment:
We may want to double check if any of these need changes in Mono or RyuJIT since they are often implemented specially in the IL importer.

CC. @jkotas

Reply:
I do not see any obvious problems. We should add some tests for these new cases.

Reply:
> We should add some tests for these new cases.

I'll add some.